test_collections/sdk_tests/support/models/matter_test_declarations.py

```diff
@@ -22,9 +22,51 @@
 )
 from test_collections.sdk_tests.support.models.sdk_test_folder import SDKTestFolder
 
-from .python_test_models import MatterTestType, PythonTest
-from .test_case import PythonTestCase
-from .test_suite import PythonTestSuite, SuiteType
+from ..python_testing.models.python_test_models import PythonTest
+from ..python_testing.models.test_case import PythonTestCase
+from ..python_testing.models.test_suite import PythonTestSuite
+from ..python_testing.models.test_suite import SuiteType as PythonSuiteType
+from ..yaml_tests.models.test_case import YamlTestCase
+from ..yaml_tests.models.test_suite import SuiteType as YamlSuiteType
+from ..yaml_tests.models.test_suite import YamlTestSuite
+from ..yaml_tests.models.yaml_test_models import YamlTest
+from .matter_test_models import MatterTestType
+
+
+class YamlCollectionDeclaration(TestCollectionDeclaration):
+    def __init__(self, folder: SDKTestFolder, name: str) -> None:
+        super().__init__(path=str(folder.path), name=name)
+        self.yaml_version = folder.version
+
+
+class YamlSuiteDeclaration(TestSuiteDeclaration):
```

Comment:
Are you sure `PythonTestSuite` can't be a subtype of a new `MatterTestSuite` along with `{Manual, Simulated, Chiptool}YamlTestSuite`?

And then have a single class `MatterSuiteDeclaration`?

```suggestion
class MatterSuiteDeclaration(TestSuiteDeclaration):
```